File: Ruby/Chap01/03.rb
str = "Now I need a drink, alcoholic of course, after the heavy lectures involving quantum mechanics."
p str.split(/\W+/).map(&:length)

###--- memo ---###
# split : 引数に正規表現を指定した場合は、そのパターンにマッチした部分で文字列を分割する
#
# /\W+/ : = [^a-zA-Z0-9_] 非単語構成文字の1回以上の繰り返し

File: Python3/Chap01/03.py
import re

s = "Now I need a drink, alcoholic of course, after the heavy lectures involving quantum mechanics."
regex = r"[^a-zA-Z0-9_]+"

l = re.split(regex, s)
l.remove("")

for i in range(0, len(l)):
    l[i] = len(l[i])

print(l)
